doc/dune-files.rst

```diff
@@ -285,6 +285,23 @@ language: The syntax is as a list of the following elements:
 
    dep-specification = dep+
 
+.. _new-foreign-flags-handling:
+
+new_foreign_flags_handling 
+--------------------------
+
+Since Dune 2.8, it is possible to deactivate
+the systematic prepending of flags coming from ``ocamlc -config`` to the C
+compiler command line. This is done adding the following field to the ``dune-project`` file:
+
+.. code:: scheme
+
+    (new_foreign_flags_handling true)
```

Comment:
I don't think that the name conveys a good description of what it does. An explicit name would be better to inform the user of what it does. And consider the possibility that there'll be a third option in the future - we don't want it to be `even_newer_flags_handling` :) For the same reason, something other than a boolean would be more future proof.

Let me suggest `(foreign_flags_type osl)` (as opposed to `(foreign_flags_type classic)`) maybe?